app/Filament/Resources/GameResource/RelationManagers/MemoryNotesRelationManager.php

```diff
@@ -0,0 +1,156 @@
+                            return "Edit Note at {$memoryNote->address_hex}";
+                        })
+                        ->modalDescription(function (MemoryNote $memoryNote): ?string {
+                            /** @var User $user */
+                            $user = auth()->user();
+
+                            if ($user->is($memoryNote->user)) {
+                                return null;
+                            }
+
+                            return "WARNING: By making changes to this note, you will become the author of the note.";
+                        })
+                        ->modalSubmitActionLabel(function (MemoryNote $memoryNote): string {
+                            /** @var User $user */
+                            $user = auth()->user();
+
+                            if ($user->is($memoryNote->user)) {
+                                return "Save changes";
+                            }
+
+                            return "Save changes and become note's author";
+                        })
+                        ->mutateFormDataUsing(function (array $data): array {
+                            /** @var User $user */
+                            $user = auth()->user();
+
+                            // The code note will always be "owned" by whoever last edited it.
+                            $data['user_id'] = $user->id;
+
+                            return $data;
+                        }),
+
+                    Tables\Actions\Action::make('delete')
+                        ->label('Delete')
+                        ->icon('heroicon-s-trash')
+                        ->color('danger')
+                        ->requiresConfirmation()
+                        ->modalHeading(function (MemoryNote $memoryNote): string {
+                            return "Delete Note at {$memoryNote->address_hex}";
+                        })
+                        ->modalDescription('Are you sure you want to delete this note? It will be irreversibly lost.')
+                        ->action(function (MemoryNote $memoryNote): void {
+                            /** @var User $user */
+                            $user = auth()->user();
+
+                            if (!$user->can('delete', $memoryNote)) {
+                                return;
+                            }
+
+                            $memoryNote->user_id = $user->id;
+                            $memoryNote->forceFill(['body' => '']);
+                            $memoryNote->save();
+                        })
+                        ->visible(function (MemoryNote $memoryNote): bool {
+                            /** @var User $user */
+                            $user = auth()->user();
+
+                            return $user->can('delete', $memoryNote);
+                        }),
+                ]),
+            ])
+            ->paginated([200])
```

Comment:
200 is a lot of items to scroll through. Maybe limit it to 100 or even 50?

Reply:
Sounds good to me. In latest I've set the default to 50 and optionally allow the user to bump up to 100.